git-sh-setup

```diff
@@ -0,0 +1,357 @@
+# This shell scriplet is meant to be included by other shell scripts
```

Comment:
Is this a script you got from somewhere else? We just need to make sure we are checking the LICENSE and it's okay to reuse this.

Reply:
its part of git and required for the subtree script to run correctly